File: testdata/bench/go/BenchmarkTest00001.go
package bench

import (
	"database/sql"
	"net/http"
)

// SQL injection via string concatenation
func Handler00001(w http.ResponseWriter, r *http.Request) {
	param := r.URL.Query().Get("id")
	query := "SELECT * FROM users WHERE id = " + param
	db, _ := sql.Open("sqlite3", ":memory:")
	db.Query(query)
}

File: testdata/bench/kotlin/BenchmarkTest00001.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via string template in Ktor
suspend fun handler00001(call: ApplicationCall) {
    val id = call.parameters["id"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    val rs = conn.createStatement().executeQuery("SELECT * FROM users WHERE id = $id")
    call.respondText(rs.toString())
}
